### ISSUE TITLE
Upgrades to v1.20.0-beta.0 causes service updates to fail with "spec.clusterIPs: Required value"

### ISSUE THREAD

**Issue by rtheis** [kubernetes/kubernetes#96057]
**What happened**:

`kubectl label service --overwrite -n kube-system kube-dns kubernetes.io/name=CoreDNS` fails with `The Service "kube-dns" is invalid: spec.clusterIPs: Required value` on Kubernetes v1.20.0-beta.0 cluster after it has been upgraded from v1.20.0-alpha.3.

**What you expected to happen**:

Command is successful.

**How to reproduce it (as minimally and precisely as possible)**:

Create a cluster at Kubernetes v1.20.0-alpha.3 with CoreDNS installed.  Upgrade the master to Kubernetes v1.20.0-beta.0.  Then run `kubectl label service --overwrite -n kube-system kube-dns kubernetes.io/name=CoreDNS`.

**Anything else we need to know?**:

See https://kubernetes.slack.com/archives/C09QYUH5W/p1603989433062500.

**Environment**:
- Kubernetes version (use `kubectl version`): Cluster created at `v1.20.0-alpha.3` and upgrade to `v1.20.0-beta.0`.  Test using client `v1.19.3`.
- Cloud provider or hardware configuration: IBM Cloud Kubernetes Service
- OS (e.g: `cat /etc/os-release`): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: Custom via IBM Cloud Kubernetes Service
- Network plugin and version (if this is a network-related bug): N/A
- Others: N/A

**Comment by aramase** [comment]
I tried testing this with an Azure cluster and was able to repro 

Steps taken - 

1. Created cluster with v1.20.0-alpha.3
```bash
➜ ./kubectl get nodes
NAME                                 STATUS   ROLES    AGE   VERSION
k8s-agentpool1-62115341-vmss000000   Ready    agent    12m   v1.20.0-alpha.3
k8s-agentpool1-62115341-vmss000001   Ready    agent    12m   v1.20.0-alpha.3
k8s-master-62115341-0                Ready    master   12m   v1.20.0-alpha.3
```

2. Tried labelling the service for kube-dns and custom service. No errors.
```bash
➜ ./kubectl label service --overwrite -n kube-system kube-dns kubernetes.io/name=CoreDNS
service/kube-dns not labeled

➜ ./kubectl label service --overwrite nginx-ipv4 test=nginx
service/nginx-ipv4 labeled
```

3. Upgraded the cluster manually -
- Updated api server to `v1.20.0-beta.0`
- Updated controller manager to `v1.20.0-beta.0`
- Updated scheduler to `v1.20.0-beta.0`
- Updated kubelet and restarted on all nodes to `v1.20.0-beta.0`

```bash

➜ ./kubectl version
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.0-beta.0", GitCommit:"f00ec1b449d2cde0f11ff71bbeebca240cd2a92b", GitTreeState:"clean", BuildDate:"2020-10-28T01:35:50Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.0-beta.0", GitCommit:"f00ec1b449d2cde0f11ff71bbeebca240cd2a92b", GitTreeState:"clean", BuildDate:"2020-10-29T18:45:02Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}

➜ ./kubectl get nodes
NAME                                 STATUS   ROLES    AGE   VERSION
k8s-agentpool1-62115341-vmss000000   Ready    agent    52m   v1.20.0-beta.0.106+3d62aad55eef8a
k8s-agentpool1-62115341-vmss000001   Ready    agent    52m   v1.20.0-beta.0.106+3d62aad55eef8a
k8s-master-62115341-0                Ready    master   52m   v1.20.0-beta.0.106+3d62aad55eef8a
```
The kubelet build is from master
		
4. Tried to label service with v1.20.0-beta.0 kubectl client and it failed 

```
➜ ./kubectl label service --overwrite -n kube-system kube-dns kubernetes.io/name=CoreDNS
The Service "kube-dns" is invalid: spec.clusterIPs: Required value

➜ ./kubectl label service --overwrite nginx-ipv4 test=nginx
The Service "nginx-ipv4" is invalid: spec.clusterIPs: Required value
```

```bash
➜ ./kubectl describe svc nginx-ipv4
Name:              nginx-ipv4
Namespace:         default
Labels:            app=nginx
                   test=nginx
Annotations:       <none>
Selector:          app=nginx
Type:              ClusterIP
IP Families:       <none>
IP:                10.0.205.108
IPs:               <none>
Port:              <unset>  80/TCP
TargetPort:        80/TCP
Endpoints:         10.244.0.4:80,10.244.0.5:80,10.244.1.4:80
Session Affinity:  None
Events:            <none>
```

**Comment by aramase** [comment]
`kubectl get svc <svc name> -o json` doesn't contain the `clusterIPs` field

```json
...
  "spec": {
    "clusterIP": "10.0.205.108",
    "ports": [
      {
        "port": 80,
        "protocol": "TCP",
        "targetPort": 80
      }
    ],
    "selector": {
      "app": "nginx"
    },
    "sessionAffinity": "None",
    "type": "ClusterIP"
  },
  "status": {
    "loadBalancer": {}
  }
```

New service created after upgrade has the `clusterIPs` field as expected

```
...
  "spec": {
    "clusterIP": "10.0.213.173",
    "clusterIPs": [
      "10.0.213.173"
    ],
    "ports": [
      {
        "port": 80,
        "protocol": "TCP",
        "targetPort": 80
      }
    ],
    "selector": {
      "app": "nginx"
    },
    "sessionAffinity": "None",
    "type": "ClusterIP"
  },
```

**Comment by aramase** [comment]
Tried an upgrade with https://github.com/kubernetes/kubernetes/pull/95894 included in the build and don't see any errors. The `ClusterIPs` field is populated on the read - https://github.com/kubernetes/kubernetes/pull/95894/files#diff-c5577c2f53301f4aa1efc47e52b01514c96642dd26d140aac54b956f99adb0ccR172-R175

**Comment by khenidak** [comment]
Thank you @aramase - I will keep this open until @rtheis confirms from his end.

**Comment by rtheis** [comment]
@khenidak I will confirm this once v1.20.0-beta.1 is released (release scheduled for today).

**Comment by rtheis** [comment]
@khenidak I have confirmed that this is fixed in v1.20.0-beta.1 via https://github.com/kubernetes/kubernetes/pull/95894.  Thank you.
